### ISSUE TITLE
Is stream column family useful to be a seperate cf instead of merging into subkey cf?

### ISSUE THREAD

**Issue by PragmaTwice** [apache/kvrocks#2330]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

As title, should we merge it to subkey cf while keeping compatible to old versions?

### Solution

- check the current cfs using `ListColumnFamilies()` before opening db:
  - if it's a new database or no stream cf:
    - don't create stream cf, use default as stream cf
  - if there's stream cf:
    - keep to use stream cf for compatibility

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by git-hulk** [comment]
@torwig @Yangsx-1 What do you think about this proposal?

**Comment by torwig** [comment]
If the only reason is compatibility, sure, we can do what @PragmaTwice offered.

**Comment by Yangsx-1** [comment]
I think the amplification of stream is not very large, it's ok to merge it to subkey cf.

**Comment by git-hulk** [comment]
> I think the amplification of stream is not very large, it's ok to merge it to subkey cf.

Good point. It might affect the reading performance if there are too many tombstones.
